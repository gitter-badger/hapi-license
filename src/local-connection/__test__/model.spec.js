import createLicenseModel from '../license.model';

describe('local connection models', () => {
    let connection;

    beforeEach(() => {
        connection = {
            define: jest.fn((table, columns) => columns)
        };
    });

    it('should create a license model', () => {
        const licenseModel = createLicenseModel(connection);
        expect(connection.define).toHaveBeenCalled();
        expect(connection.define).toHaveBeenCalledTimes(1);
        expect(licenseModel).toMatchSnapshot();
    });
});
