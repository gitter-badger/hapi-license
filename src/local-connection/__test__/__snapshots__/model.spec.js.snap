// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`local connection models should create a license model 1`] = `
Object {
  "accountId": Object {
    "allowNull": false,
    "type": [Function],
    "validate": Object {
      "notEmpty": true,
    },
  },
  "computerId": Object {
    "allowNull": false,
    "type": [Function],
    "validate": Object {
      "notEmpty": true,
    },
  },
  "computerName": Object {
    "allowNull": false,
    "type": [Function],
    "validate": Object {
      "notEmpty": true,
    },
  },
  "computerOS": Object {
    "allowNull": false,
    "type": [Function],
    "validate": Object {
      "isIn": Array [
        Array [
          "Windows",
          "MacOS",
          "Linux",
        ],
      ],
    },
  },
  "computerUsername": Object {
    "allowNull": false,
    "type": [Function],
    "validate": Object {
      "notEmpty": true,
    },
  },
  "id": Object {
    "defaultValue": UUIDV4 {},
    "primaryKey": true,
    "type": [Function],
  },
  "license": Object {
    "allowNull": false,
    "type": [Function],
  },
  "productId": Object {
    "allowNull": false,
    "type": [Function],
    "validate": Object {
      "notEmpty": true,
    },
  },
  "validUntil": Object {
    "allowNull": false,
    "type": [Function],
  },
}
`;
